### ISSUE TITLE
Always receiving a BLE count of zero

### ISSUE THREAD

**Issue by zapoig** [dbSuS/libpax#17]
## Environment :
- Windows 10 (also reproduced on Linux)
- Arduino IDE 1.8.19
- ESP32 espressif arduino boards packages 1.0.6
- T-Beam v1.0 dev board (reproduced on many different other boards)

## Steps to reproduce :
- clone libpax branch master
- edit _lib/libpax/libpax_api.h_ to uncomment build-time options _LIBPAX_WIFI_ and _LIBPAX_WBLE_
- copy _lib/libpax_ in arduino libraries folder
- copy the sample program _src/main.cpp_ as _src/main.ino_ and open it in arduino ide
- modify it like this (removing non-arduino version for clarity, and defining _LIBPAX_ARDUINO_ as instructed in README.md) :
```
#define LIBPAX_ARDUINO
#include <libpax_api.h>
#include <stdio.h>

struct count_payload_t count_from_libpax;

void process_count(void) {
  printf("pax: %d; %d; %d;\n", count_from_libpax.pax, count_from_libpax.wifi_count, count_from_libpax.ble_count);
}

void init() {
  struct libpax_config_t configuration; 
  libpax_default_config(&configuration);
  configuration.blecounter = 1;
  configuration.blescantime = 0; // infinit
  configuration.wificounter = 1; 
  configuration.wifi_channel_map = WIFI_CHANNEL_ALL;
  configuration.wifi_channel_switch_interval = 50;
  configuration.wifi_rssi_threshold = -80;
  configuration.ble_rssi_threshold = -80;
  libpax_update_config(&configuration);

  // internal processing initialization
  libpax_counter_init(process_count, &count_from_libpax, 10, 1); 
  libpax_counter_start();
}

void setup() {
  init();
}

void loop() {}
```
- build and upload, open the serial monitor.

## Expected output:
Receiving values for both wifi and ble counts.

## Actual output:
16:25:09.268 -> pax: 3; 3; 0;
... 10 minutes later
16:25:19.262 -> pax: 7; 7; 0;
BLE count is always zero.

Note that running the BLE_scan sample sketch included with ESP32 arduino package finds around 15 BLE devices at the same location, same time on the same board.

**Comment by zapoig** [comment]
All tests are also successful with `pio test -e espidf`.

**Comment by cyberman54** [comment]
Tests show BLE sniffing results (see third column) as expected, thus BLE sniffing works.

Looks like in integration / API problem with your app. Did you set the build time options in your app?

```
#define LIBPAX_WIFI // enables WiFi sniffing features in build 
#define LIBPAX_BLE  // enables BLE sniffing features in build
```

**Comment by zapoig** [comment]
At the moment I am not using any code of my own, I'm testing with only the libpax provided sample "src/main.cpp"

As described in my initial post I have uncommented the build time options directly in the libpax file _libpax_api.h_ so they are always enabled.

**Comment by cyberman54** [comment]
In the example there is an RSSI threshold set:

```
configuration.wifi_rssi_threshold = -80;
configuration.ble_rssi_threshold = -80;
```

Perhaps your BLE signal do not hit the threshold. Please set to zero and try again.

**Comment by cyberman54** [comment]
I installed Arduino IDE 1.8.19, compiled and uploaded on a ESP32 T-Beam Board, an can reproduce this issue:

**Comment by zapoig** [comment]
That's similar to my own results. Thanks for your time testing that.

**Comment by zapoig** [comment]
I finally succeeded in building the main.cpp sample with esp-idf v4.4, and the BLE counting works there.

**Comment by cyberman54** [comment]
@zapoig thanks for the note. 
We should leave this open, until it's clear why it does not work with Arduino IDE.

**Comment by bleeptrack** [comment]
I'm also experimenting with libpax and ran into the same issue:
Building with platform.io and this configuration
```
[env:nodemcu-32s]
platform = espressif32
board = nodemcu-32s
framework = arduino
```
gives me 0 BLE devices.

**Comment by zapoig** [comment]
> I'm also experimenting with libpax and ran into the same issue: Building with platform.io and this configuration
> 
> ```
> [env:nodemcu-32s]
> platform = espressif32
> board = nodemcu-32s
> framework = arduino
> ```
> 
> gives me 0 BLE devices.

Did you try to run the tests ?
`pio test -e arduino`

**Comment by zapoig** [comment]
@lifezoned4 Yes I had tried setting rssi_thresholds (for both wifi and ble) to zero before.
Here is the results I have today :

When unit tests yield counts :
```
libpax should be running
pax: 18; 1; 17;
pax: 21; 1; 20;
pax: 21; 1; 20;
pax: 21; 1; 20;
pax: 23; 1; 22;
pax: 23; 1; 22;
E (41212) event: Event loop not initialized via esp_event_loop_init, but esp_event_send called
after stop: Current free heap: 208120
libpax should be stopped
```

The sample, built with arduino IDE, only finds wifis with default RSSIs = -80 :
```
pax: 1; 1; 0;
pax: 1; 1; 0;
pax: 2; 2; 0;
pax: 2; 2; 0;
pax: 3; 3; 0;
pax: 3; 3; 0;
```

And - still building using arduino IDE - switching the RSSIs to 0 doesn't change the BLE zero count (although it appears to find more wifis) :
```
pax: 4; 4; 0;
pax: 7; 7; 0;
pax: 8; 8; 0;
pax: 11; 11; 0;
pax: 14; 14; 0;
pax: 15; 15; 0;
```

Now if I build the original sample (rssi at -80) with platformio, it is finding both ble and wifi nodes :
```
pax: 6; 1; 5;
pax: 6; 1; 5;
pax: 6; 1; 5;
pax: 6; 1; 5;
pax: 6; 1; 5;
pax: 8; 1; 7;
```

And switching the rssi to 0 improves both counts - with platformio build again :
```
pax: 16; 2; 14;
pax: 17; 2; 15;
pax: 17; 2; 15;
pax: 17; 2; 15;
pax: 20; 2; 18;
pax: 21; 2; 19;
```

In the end I am happy having switched to platformio for my own project, but if the arduino IDE support is broken it should probably get mentioned in the README.

**Comment by cyberman54** [comment]
@zapoig thanks for the report. It's strange, i have no idea what's the root cause here.
